sdl2promela/translator.py

```diff
@@ -1046,19 +1047,23 @@ def __generate_assignment(
             context, statements, left, finalType, 0
         )
 
-        for index in range(int(finalType.Max)):
-            element = (
-                ArrayAccessBuilder()
-                .withArray(__generate_variable_name(context, data_field, True))
-                .withIndex(promelamodel.IntegerValue(index))
-                .build()
-            )
-            statements.append(
-                AssignmentBuilder()
-                .withTarget(element)
-                .withSource(promelamodel.IntegerValue(0))
-                .build()
-            )
+        if (
+            finalType.kind == OCTET_STRING_TYPE_NAME
+            or finalType.kind == IA5_STRING_TYPE_NAME
+        ):
+            for index in range(int(finalType.Max)):
```

Comment:
I think that I missed this in previous MRs.... we do we need a separate assignment function, and while do we explicitly generate a[0].... a[max] instead of for (i : 0 ... max) a[i]?

Reply:
Ok, on the left side is an variable, on the right side is a value. `data := {0, 1, 2}`. So in promela this should looks like:
` data.data[0] = 0;
data.data[1] = 1;
data.data[2] = 2;
data.length = 3;
`
There's no source variable to iterate in promela code